src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -1034,6 +1051,91 @@ private void rejectNorthernIrelandHouseholdToCE(
     }
   }
 
+  private void rejectIfForCrownDependency(String postcode) throws CTPException {
+    String postcodeArea = postcode.substring(0, 2).toUpperCase();
+
+    switch (postcodeArea) {
+      case "GY":
+      case "JE":
+        log.with(postcode).info("Rejecting request as postcode is for a channel island address");
+        throw new CTPException(
+            Fault.BAD_REQUEST, "Channel Island addresses are not valid for Census");
+      case "IM":
+        log.with(postcode).info("Rejecting request as postcode is for an Isle of Man address");
+        throw new CTPException(Fault.BAD_REQUEST, "Isle of Man addresses are not valid for Census");
+      default: // to keep checkstyle happy
+    }
+  }
+
+  private uk.gov.ons.ctp.integration.contactcentresvc.representation.Region determineActualRegion(
+      NewCaseRequestDTO caseRequestDTO) throws CTPException {
+    uk.gov.ons.ctp.integration.contactcentresvc.representation.Region requestedRegion =
+        caseRequestDTO.getRegion();
+    uk.gov.ons.ctp.integration.contactcentresvc.representation.Region actualRegion = null;
+
+    String postcode = caseRequestDTO.getPostcode();
+    String postcodeArea = postcode.substring(0, 2).toUpperCase();
+
+    if (postcodeArea.equals("BT")) {
+      log.with(postcode).debug("Forcing region to Northern Ireland");
+      actualRegion = uk.gov.ons.ctp.integration.contactcentresvc.representation.Region.N;
+
+    } else {
+      // Get ready to call AI to find the region for the specified postcode
```

Comment:
As discussed I would have considered calling the existing code for search by Postcode (`AddressServiceClientServiceImpl.searchByPostcode`) to save a bit of coding.

Reply:
I looked at this class again and agree that I should have done.
I'd update if it wasn't a rushed change

Reply:
Agree, but at least deviation here has used limit=1
Hey ho - we throw this code away in a month or so